Extensions/MGLMapView+Example.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+import Mapbox
+import Turf
+
+extension MGLMapView {
+    func coordinateBoundsInset(_ inset: CGSize) -> MGLCoordinateBounds {
+        return convert(bounds.insetBy(dx: inset.width, dy: inset.height), toCoordinateBoundsFrom: nil)
```

Comment:
`MGLMapView.convert(_:toCoordinateBoundsFrom:)` is inaccurate on a tilted map: mapbox/mapbox-gl-native#2259.

More broadly, I think we should investigate positioning the labels based on screen coordinates rather than geographic coordinates. Core GL’s own symbol collision avoidance code is based on projected coordinates, which account for rotation and tilting, so it would make sense to be consistent with core GL labeling on that respect.

Imagine taking the maneuver coordinates for a given route, converting them all to screen points, and stringing them together in a UIBezierPath. Calculating the distance between two points is suddenly just a Pythagorean distance problem.

Reply:
Removed this extension since I no longer needed it. I am still doing a lot of my calculations in geographic coordinates, but do filter them to only use ones that are onscreen. Also, when coordinates are chosen for the annotations I use the on-screen position to decide where to put the "stem" of the bubble callout.